api/v1alpha1/akodeploymentconfig_webhook_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+    "testing"
+
+    . "github.com/onsi/gomega"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAdcControlPlaneNetworkImmutable(t *testing.T) {
+    tests := []struct {
+        name           string
+        oldControlPlane ControlPlaneNetwork
+        newControlPlane ControlPlaneNetwork
+        expectErr      bool
+    }{
+        {
+            name:           "when the control plane name and CIDR is not changed",
+            oldControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 1",
+                CIDR: "10.1.0.0/24",
+            },
+            newControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 1",
+                CIDR: "10.1.0.0/24",
+            },
+            expectErr:      false,
+        },
+        {
+            name:           "when the old control plane is empty",
+            oldControlPlane: ControlPlaneNetwork{},
+            newControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 1",
+                CIDR: "10.1.0.0/24",
+            },
+            expectErr:      true,
+        },
+        {
+            name:           "when the control plane name is changed",
+            oldControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 1",
+                CIDR: "10.1.0.0/24",
+            },
+            newControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 2",
+                CIDR: "10.1.0.0/24",
+            },
+            expectErr:      true,
+        },
+        {
+            name:           "when the control plane CIDR is changed",
+            oldControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 1",
+                CIDR: "10.1.0.0/24",
+            },
+            newControlPlane: ControlPlaneNetwork{
+                Name: "VM Network 1",
+                CIDR: "10.0.0.0/24",
+            },
+            expectErr:      true,
+        },
```

Comment:
just nit:  add a test case when the newControlPlane is empty

Reply:
yeah, it's good idea